### ISSUE TITLE
My Challenges Bucket: Must include challenges the user is a resource of (eg copilot, reviewer etc)

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4602]
My Challenges Bucket: Must include challenges the user is a resource of (eg copilot, reviewer etc)

Example:
user picachui is a reviewer for challenge
http://www.topcoder.com/challenge-details/30131098
https://beta-community-app.topcoder.com/challenges/d9b51702-711f-4ee4-93ab-5af4271e83f3
user picachui is copilot for challenge
http://www.topcoder.com/challenge-details/30130833
https://beta-community-app.topcoder.com/challenges/0979eb93-c8b0-44ae-95a9-0f927ed5ee4c

These challenges are displayed under the My Challenges bucket in V4, but not in V5

<img width="1440" alt="Screenshot 2020-06-25 at 12 56 50 PM" src="https://user-images.githubusercontent.com/58783823/85678575-3ec43200-b6e6-11ea-99ed-223a76b2b034.png">
<img width="1440" alt="Screenshot 2020-06-25 at 12 57 04 PM" src="https://user-images.githubusercontent.com/58783823/85678608-45eb4000-b6e6-11ea-8fda-a8496824f8cf.png">

**Comment by sushilshinde** [comment]
@rootelement please guide on this

**Comment by rootelement** [comment]
@sushilshinde @SathyaJayabal , i _believe_ this is due to an issue we had with the legacy-challenge-resource-processor. Can you please re-test? 

Because Picachui is not showing up on the API side here either, so that means their username is not associated with the challenge.
http://api.topcoder-dev.com/v5/resources?challengeId=d72fd406-f08c-441f-b44f-13316acb94da
<img width="1480" alt="Screenshot 2020-07-02 at 8 45 45 PM" src="https://user-images.githubusercontent.com/5400156/86393469-1f3f8300-bc6b-11ea-9f33-5fb35ccde42f.png">

**Comment by SathyaJayabal** [comment]
@rootelement @sushilshinde ,

Lets take this example:
https://beta-community-app.topcoder.com/challenges/d9b51702-711f-4ee4-93ab-5af4271e83f3
https://www.topcoder.com/challenges/30131098

User picachui is a Reviewer
<img width="1440" alt="Screenshot 2020-07-03 at 9 29 37 AM" src="https://user-images.githubusercontent.com/58783823/86430381-bbbb6100-bd0f-11ea-8b22-23d4264f2064.png">

But the resources api call does not return this user
<img width="840" alt="Screenshot 2020-07-03 at 9 29 59 AM" src="https://user-images.githubusercontent.com/58783823/86430416-d8579900-bd0f-11ea-8c64-bbc5c2f99ed6.png">

And this challenge does not show up in the My Challenges bucket in V5(beta)
<img width="1440" alt="Screenshot 2020-07-03 at 9 32 03 AM" src="https://user-images.githubusercontent.com/58783823/86430565-3e442080-bd10-11ea-8eb6-a6b633c80740.png">

It must show up in My challenges bucket like in V4(Prod)
<img width="1440" alt="Screenshot 2020-07-03 at 9 32 33 AM" src="https://user-images.githubusercontent.com/58783823/86430586-4ef49680-bd10-11ea-87c8-33a344a9391c.png">


In Summary: the resources api call is only returning users with submitter roles
<img width="1440" alt="Screenshot 2020-07-03 at 9 39 30 AM" src="https://user-images.githubusercontent.com/58783823/86431012-3cc72800-bd11-11ea-9256-fe356d9d8e0c.png">

**Comment by mtwomey** [comment]
## Regarding adding reviewers in OR and having them show up to the V5 API
@SathyaJayabal the resources api call in the browser is unauthenticated. The unauthenticated call will only return submitters. For this call to return other resources, it needs to be passed an m2m token. We confirmed that the same request, using Postman and an m2m token does return all the resources (including you as a reviewer).

**Example Curl for reference**
```
curl --location --request GET 'https://api.topcoder.com/v5/resources?challengeId=d9b51702-711f-4ee4-93ab-5af4271e83f3' \
--header 'Authorization: Bearer [M2M TOKEN]'
```
**Results (truncated)**
[note: `318b9c07-079a-42d9-a81f-b96be1dc1099` is the UUID for a reviewer]
```
    {
        "updated": "2020-06-25T03:38:30.000Z",
        "challengeId": "d9b51702-711f-4ee4-93ab-5af4271e83f3",
        "created": "2020-06-25T03:38:30.000Z",
        "memberId": "22778049",
        "updatedBy": "picachui",
        "id": "6108334f-f5aa-4e08-bce2-964ac8be3d03",
        "createdBy": "picachui",
        "memberHandle": "picachui",
        "roleId": "318b9c07-079a-42d9-a81f-b96be1dc1099",
        "rating": 1036
    },
```
----
As a second test we added my account (mtwomey) to this test challenge using v5-api:

**Curl**
```
curl --location --request POST 'https://api.topcoder.com/v5/resources' \
--header 'Authorization: Bearer [M2M TOKEN]' \
--header 'Content-Type: application/json' \
--data-raw '{
	"challengeId": "d9b51702-711f-4ee4-93ab-5af4271e83f3",
	"memberHandle": "mtwomey",
	"roleId": "318b9c07-079a-42d9-a81f-b96be1dc1099"
}'
```

Adding myself worked, however at this point we found that the challenge didn't sync back to v4. This is the legacy-challenge-resource-processor (https://console.aws.amazon.com/ecs/home?region=us-east-1#/clusters/tc-challenge-legacy/tasks).

Found some issues with the legacy processor which is responsible for updating v4 when v5 resources are changed. This is likely not related to this issue, just noted here. Not required for Community-app at this time (opened a ticket: https://github.com/appirio-tech/ap-challenge-microservice/issues/542)

----
As a third test, we added myself as a reviewer in OR and I do show up as a reviewer (via an _authenticated_ call to https://api.topcoder.com/v5/resources?challengeId=d9b51702-711f-4ee4-93ab-5af4271e83f3). **Note**: It takes approximately 3 - 5 minutes for this to show up (to V5) due to ES, processors, kafka, events, ...etc

## Regarding a user who is a reviewer on a challenge - that challenge should show up in their "My Challenges List"

See James note on this in slack, he believe he's found the issue and is correcting.

**Comment by SathyaJayabal** [comment]
cc  @rootelement

**Comment by mtwomey** [comment]
Ok - @rootelement 's fix has corrected this issue. **Note** It still takes a long time to update because the 9 **_authenticated_** calls to: https://api.topcoder.com/v5/challenges/?status=Active&perPage=99&page=2 take a long time each. That is a separate issue. @SathyaJayabal Kindly re-test

**Comment by SathyaJayabal** [comment]
@mtwomey @rootelement ,  I was able  to see the above challenge under My Challenges bucket for user picachui, where picachui is a reviewer.

But there are several other examples where this doesn't work as expected:
1. picachui user is a manager for the the below contest (for past many days). 
https://beta-community-app.topcoder.com/challenges/dfb8c1e0-3173-4150-987c-3fea61da2548
https://www.topcoder.com/challenges/30130833

This challenge must be displayed under "My Challenges" like in Prod( V4)
<img width="1440" alt="Screenshot 2020-07-10 at 12 19 23 PM" src="https://user-images.githubusercontent.com/58783823/87125153-a44c1d00-c2a7-11ea-9563-b63f8f71feb2.png">

<img width="1440" alt="Screenshot 2020-07-10 at 12 19 32 PM" src="https://user-images.githubusercontent.com/58783823/87125165-aada9480-c2a7-11ea-959a-9e88a41ec066.png">

2. Removed user picachui as reviewer from the below  challenge in OR
https://www.topcoder.com/challenges/30131098
https://beta-community-app.topcoder.com/challenges/0ea10b6f-c089-457b-8c08-b01be006b25e

The challenge was removed from My Challenges bucket in Prod(v4), but not in V5 even after 40 mins
<img width="1440" alt="Screenshot 2020-07-10 at 12 23 57 PM" src="https://user-images.githubusercontent.com/58783823/87125496-43711480-c2a8-11ea-8469-2912f4587cec.png">
<img width="1440" alt="Screenshot 2020-07-10 at 12 24 01 PM" src="https://user-images.githubusercontent.com/58783823/87125533-4d931300-c2a8-11ea-8225-2b20823c8a9d.png">

3. This is an example in dev env. There are a lot of challenges missing from My challenge in dev env. This is one example.
user: jcori is manager for three wireframe contests. But only one of them show up in V5 under My Challenges
https://test-community-app.topcoder-dev.com/challenges/ce86b0a1-04f5-4ce7-a962-3602c5a7b909
https://test-community-app.topcoder-dev.com/challenges/1d757c5b-6614-4ca6-a2c6-cf59a239424f
https://test-community-app.topcoder-dev.com/challenges/205aef7e-b52a-40e1-ad43-f1a0fc984af6

<img width="1440" alt="Screenshot 2020-07-10 at 11 23 59 AM" src="https://user-images.githubusercontent.com/58783823/87125785-b37f9a80-c2a8-11ea-97ea-01bdd615425a.png">
<img width="1440" alt="Screenshot 2020-07-10 at 11 24 06 AM" src="https://user-images.githubusercontent.com/58783823/87125822-c09c8980-c2a8-11ea-96e2-61dc80983a36.png">

cc @lakshmiathreya

**Comment by mtwomey** [comment]
Basically - this works for being a reviewer, but it doesn't work properly for some other roles. Known roles where this isn't working:

* manager

Known working

* Reviewer
* Observer
* Submitter

@rootelement This is going to be that same fix we did for reviewer on Thursday...

**Comment by SathyaJayabal** [comment]
@mtwomey @sushilshinde @lakshmiathreya 
The my challenges displays all the challenges the user is a resource of except for some users (tracked here #4642).

**Comment by lakshmiathreya** [comment]
Code freeze Re-testing: verified with Submitter, Observer and Approver roles. Will test with the other roles once Support enables those for my handle.

**Comment by lakshmiathreya** [comment]
@SathyaJayabal - pls added "CF Retested" label after you verify for Copilot, Manager and all Reviewer roles.

**Comment by SathyaJayabal** [comment]
Retested for Manager,Copilot and Reviewer roles
